bundles/org.openhab.binding.ferroamp/src/main/java/org/openhab/binding/ferroamp/internal/FerroampHandler.java

```diff
@@ -0,0 +1,255 @@
+        String valueConfiguration = command.toString();
+        if (FerroampBindingConstants.CHANNEL_REQUESTCHARGE.equals(channelUID.getId())) {
+            String requestCmdJsonCharge = "{\"" + "transId" + "\":\"" + transId
+                    + "\",\"cmd\":{\"name\":\"charge\",\"arg\":\"" + valueConfiguration + "\"}}";
+            FerroampMqttCommunication.sendPublishedTopic(requestCmdJsonCharge, ferroampConfig);
+        }
+        if (FerroampBindingConstants.CHANNEL_REQUESTDISCHARGE.equals(channelUID.getId())) {
+            String requestCmdJsonDisCharge = "{\"" + "transId" + "\":\"" + transId
+                    + "\",\"cmd\":{\"name\":\"discharge\",\"arg\":\"" + valueConfiguration + "\"}}";
+            FerroampMqttCommunication.sendPublishedTopic(requestCmdJsonDisCharge, ferroampConfig);
+        }
+        if (FerroampBindingConstants.CHANNEL_AUTO.equals(channelUID.getId())) {
+            String requestCmdJsonAuto = "{\"" + "transId" + "\":\"" + transId + "\",\"cmd\":{\"name\":\"auto\"}}";
+            FerroampMqttCommunication.sendPublishedTopic(requestCmdJsonAuto, ferroampConfig);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        // Set configuration parameters
+        ferroampConfig = getConfigAs(FerroampConfiguration.class);
+
+        // Set channel configuration parameters
+        channelConfigEhub = FerroampChannelConfiguration.getChannelConfigurationEhub();
+        channelConfigSsoS1 = FerroampChannelConfiguration.getChannelConfigurationSsoS1();
+        channelConfigSsoS2 = FerroampChannelConfiguration.getChannelConfigurationSsoS2();
+        channelConfigSsoS3 = FerroampChannelConfiguration.getChannelConfigurationSsoS3();
+        channelConfigSsoS4 = FerroampChannelConfiguration.getChannelConfigurationSsoS4();
+        channelConfigEso = FerroampChannelConfiguration.getChannelConfigurationEso();
+        channelConfigEsm = FerroampChannelConfiguration.getChannelConfigurationEsm();
+
+        ferroampConfig = getConfigAs(FerroampConfiguration.class);
+        if (!ferroampConfig.hostName.isEmpty() || !ferroampConfig.password.isEmpty()
+                || !ferroampConfig.userName.isEmpty()) {
+            final MqttBrokerConnection ferroampConnection = new MqttBrokerConnection(ferroampConfig.hostName,
+                    FerroampBindingConstants.BROKER_PORT, false, false, ferroampConfig.userName);
+            scheduler.scheduleWithFixedDelay(this::pollTask, 60, refreshInterval, TimeUnit.SECONDS);
+            this.setFerroampConnection(ferroampConnection);
+            updateStatus(ThingStatus.UNKNOWN);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        }
+    }
+
+    private void pollTask() {
+        try {
+            startMqttConnection(getConfigAs(FerroampConfiguration.class));
+        } catch (InterruptedException e) {
+            logger.debug("Not connected to the MqttBroker");
+            return;
```

Comment:
The Thing state should also be set to something like: offline / communication error / "Connection interrupted"

Reply:
20250319
Line 98, removed
Line 99, changed, now line 96
Moved line 105 to line 100

For Thing state, Please see line 112